include/uio.h

```diff
@@ -21,6 +22,20 @@ typedef struct uio {
   vm_map_t *uio_vmspace; /* destination address space */
 } uio_t;
 
+/* Convenience function for initializing uio, where iov has one buffer */
+void prepare_single_uio(uio_t *uio, iovec_t *iov, uio_op_t op, vm_map_t *vm_map,
+                        size_t offset, void *buffer, size_t buflen);
+
+#define prepare_single_user_uio(uio, op, offset, buffer, buflen)               \
+  iovec_t __CONCAT(iov_, __LINE__);                                            \
+  prepare_single_uio((uio), &__CONCAT(iov_, __LINE__), (op),                   \
+                     get_user_vm_map(), (offset), (buffer), (buflen));
+
+#define prepare_single_kernel_uio(uio, op, offset, buffer, buflen)             \
+  iovec_t __CONCAT(iov_, __LINE__);                                            \
```

Comment:
In use cases such as `tests/vfs.c` structure `iovec_t` is defined multiple times within the same scope. Could you check if compiler optimizes that and allocates place on stack only once?